src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -243,6 +243,42 @@ bool ParserIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 }
 
 
+bool ParserTableAsStringLiteralIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    if (pos->type != TokenType::StringLiteral)
+        return false;
+
+    ReadBufferFromMemory in(pos->begin, pos->size());
+    String s;
+
+    try
+    {
+        readQuotedStringWithSQLStyle(s, in);
```

Comment:
We have `tryReadQuotedStringInto` - we can avoid throwing and catching exceptions here.